### PR TITLE
HHH-12530 -  Add processing of unknown hints

### PR DESCRIPTION
**Pull request by schernolyas**
https://hibernate.atlassian.net/browse/HHH-12530

### CODE REVIEW

**Comment by Naros** [comment]
@schernolyas There seems to be a number of tab/indent changes in your commit.  Could you fix that so only your change appears in the diff.

**Comment by schernolyas** [comment]
HI @Naros !

I have restored formatting. YOu can continue review.
Thanks!

**Comment by Naros** [comment]
@schernolyas Much thanks!

File: hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java [hunk 1]
```diff
@@ -1103,12 +1103,16 @@ else if ( QueryHints.HINT_PASS_DISTINCT_THROUGH.equals( hintName ) ) {
 		}
 
 		if ( !applied ) {
-			MSG_LOGGER.debugf( "Skipping unsupported query hint [%s]", hintName );
+			handleUnrecognizedHint(hintName, value);
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java]: It should be `handleUnrecognizedHint( hintName, value );` with space padding.

File: hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java [hunk 2]
```diff
@@ -1103,12 +1103,16 @@ else if ( QueryHints.HINT_PASS_DISTINCT_THROUGH.equals( hintName ) ) {
...
 		}
 
 		return this;
 	}
 
+	protected void handleUnrecognizedHint(String hintName,Object value) {
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/query/internal/AbstractProducedQuery.java]: Please add a space after the comma (this is the one that made me comment as it's quite annoying).

Thanks!

**Comment by Naros** [comment]
Applied upstream & fixed the syntax issue myself.

**Comment by sebersole** [comment]
@Naros please close the corresponding Jira when you integrate these.  Thanks

**Comment by sebersole** [comment]
Oops, got directed to the wrong PR.  Sorry

**Comment by Naros** [comment]
@sebersole no worries. :)

**Pull request closed without merging**